components/brave_new_tab_ui/helpers/scrolling.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react'
+
+const overflowScrollableRegex = /(auto)|(scroll)/g
+const isScrollable = (element: Element) => {
+    return overflowScrollableRegex.test(getComputedStyle(element).overflow)
+}
+
+interface Options {
+    limit: number
+}
+
+export const getScrollableParents = (element: Element | null | undefined, options?: Options) => {
+    const scrollableElements: Element[] = []
+
+    while (element && (!options?.limit || scrollableElements.length < options.limit)) {
+        if (isScrollable(element)) { scrollableElements.push(element) }
+        element = element.parentElement
+    }
+
+    return scrollableElements
+}
+
+export const getScrollableParent = (element: Element | null | undefined) => getScrollableParents(element, { limit: 1 })[0]
+
+export const useParentScrolled = (ref: React.MutableRefObject<HTMLElement | undefined>, handler: (e: Event) => void) => {
+    React.useEffect(() => {
+        const scrollable = getScrollableParent(ref.current)
+        if (!scrollable) return
+
+        scrollable.addEventListener('scroll', handler)
+        return () => {
+            scrollable.removeEventListener('scroll', handler)
+        }
+    }, [handler])
```

Comment:
does this not need to be re-run if `ref.current` changes?

Reply:
Hmm, I don't think it's as simple as add `ref.current` to the deps list, as the effect might not get rerun when the ref changes (because it's a ref).

I think I'll just store the ref in state one level up, to ensure we have an up to date value.